frontend/src/component/AddSpecies.tsx

```diff
@@ -0,0 +1,68 @@
+import {Habitat, NewSpecies} from "../model/Species";
+import {ChangeEvent, useState} from "react";
+
+type AddSpeciesProps = {
+    addSpecies: (newSpecies: NewSpecies) => Promise<any>;
+}
+
+export default function AddSpecies(props: AddSpeciesProps) {
+
+    const [name, setName] = useState<string>("");
+    const [waterDemand, setWaterDemand] = useState<number>(5);
+    const [fertilizerDemand, setFertilizerDemand] = useState<number>(5);
+    const [habitat, setHabitat] = useState<number>(1);
+    const [soil, setSoil] = useState<number>(1);
```

Comment:
Könnte man das nicht einfach in einen gemeinsamen State (zB "Plant") packen und dann mit eine gemeinsamen handleChange Funktion updaten?